pkg/internal/tracking/query.go

```diff
@@ -0,0 +1,28 @@
+package tracking
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+)
+
+func AppendMetadataToSql(sql string, metadata Metadata) (string, error) {
+	bytes, err := json.Marshal(metadata)
+	if err != nil {
+		return "", fmt.Errorf("failed to marshal the metadata: %w", err)
+	} else {
+		return fmt.Sprintf("%s --%s %s", sql, MetadataPrefix, string(bytes)), nil
+	}
+}
+
+func ParseMetadataFromSql(sql string) (Metadata, error) {
+	parts := strings.Split(sql, fmt.Sprintf("--%s", MetadataPrefix))
+	if len(parts) != 2 {
+		return Metadata{}, fmt.Errorf("failed to parse metadata from sql, incorrect number of parts, expected: 2, got: %d", len(parts))
+	}
+	var metadata Metadata
+	if err := json.Unmarshal([]byte(strings.TrimSpace(parts[1])), &metadata); err != nil {
```

Comment:
Nit: I think that Unmarshal doesn't require the data to be trimmed.

Reply:
I would leave it just to be sure